Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/NolPayLinkedCardsComponent.swift

```diff
@@ -13,25 +13,53 @@ public class NolPayLinkedCardsComponent: PrimerHeadlessComponent {
 
     var nolPay: PrimerNolPayProtocol?
 
-    public var errorDelegate: PrimerHeadlessErrorableDelegate?
+    public weak var errorDelegate: PrimerHeadlessErrorableDelegate?
     public weak var validationDelegate: PrimerHeadlessValidatableDelegate?
     var mobileNumber: String?
     var countryCode: String?
     var phoneMetadataService: NolPayPhoneMetadataProviding?
+    var apiClient: PrimerAPIClientProtocol?
 
-    public init() {
+    public init() {}
+
+    public func getLinkedCardsFor(mobileNumber: String, completion: @escaping (Result<[PrimerNolPaymentCard], PrimerError>) -> Void) {
+
+        let dispatchGroup = DispatchGroup()
+        dispatchGroup.enter()  // Enter the dispatch group
+
+        // Start the initial setup
+        start { [weak self] result in
+            defer {
+                dispatchGroup.leave() // Leave the dispatch group when done
+            }
+
+            switch result {
+            case .success:
+                break
+            case .failure(let error):
+                self?.errorDelegate?.didReceiveError(error: error)
+                completion(.failure(error))
+                return
+            }
+        }
+
+        dispatchGroup.notify(queue: .main) { [weak self] in
+            // This block is executed once the start function is complete
+            self?.continueWithLinkedCardsFetch(mobileNumber: mobileNumber, completion: completion)
+        }
+    }
+
+    private func start(completion: @escaping (Result<Void, PrimerError>) -> Void) {
         guard let nolPaymentMethodOption = PrimerAPIConfiguration.current?.paymentMethods?.first(where: { $0.internalPaymentMethodType == .nolPay})?.options as? MerchantOptions,
               let appId = nolPaymentMethodOption.appId
         else {
             let error = PrimerError.invalidValue(key: "Nol AppID",
                                                  value: nil,
-                                                 userInfo: [
-                                                "file": #file,
-                                                "class": "\(Self.self)",
-                                                "function": #function,
-                                                "line": "\(#line)"
-                                            ],
-                                            diagnosticsId: UUID().uuidString)
+                                                 userInfo: ["file": #file,
+                                                            "class": "\(Self.self)",
+                                                            "function": #function,
+                                                            "line": "\(#line)"],
+                                                 diagnosticsId: UUID().uuidString)
             ErrorHandler.handle(error: error)
             self.errorDelegate?.didReceiveError(error: error)
             return
```

Comment:
Not 100% sure on this but will suggest anyway: should `nolPay` be set to `nil` in each of these guards?

Reply:
Not sure what am I getting with this?